tangos/input_handlers/halo_stat_files/__init__.py

```diff
@@ -56,22 +57,29 @@ def all_columns(self):
     def iter_rows_raw(self, *args):
         """
         Yield the halo ID and requested columns from each line of the stat file, without any emulation.
-
+        If _id_from_file_pos is True, the ID is not taken from the stat file, but rather from the order in which
+        the halos appear.
         :param args: strings for the column names
         :return: id, arg1, arg2, arg3 where ID is the halo ID and argN is the value of the Nth named column
         """
         with open(self.filename) as f:
             header = self._read_column_names(f)
+            cnt = 0
             ids = [0]
             for a in args:
                 try:
                     ids.append(header.index(a))
                 except ValueError:
                     ids.append(None)
-
             for l in f:
                 if not l.startswith("#"):
-                    yield self._get_values_for_columns(ids, l)
+                    col_data = self._get_values_for_columns(ids, l)
+                    col_data.insert(0, cnt+self._catalog_index_offset)
```

Comment:
that's because the docstring is not accurate! I will fix this. Now iter_rows always returns the catalog index (+ any offset set by the class) AND the raw finder ID number read directly from the file, followed by all of the requested quantities.

Reply:
I've updated the docstring for the iter_rows_raw function. Hopefully it is more clear